### PR TITLE
Cbrash/city optional

### PR DESCRIPTION
**Pull request by colinbrash**
**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

**Post-merge:**
- [ ] [Backported](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2065367719/Making+a+pull+request+for+a+named+release) to latest and next named releases

### CODE REVIEW

File: sanctions/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -119,3 +119,32 @@ def test_sdn_check_search_succeeds_despite_DB_issue(
...
+    @mock.patch('sanctions.apps.api.v1.views.checkSDNFallback')
+    @mock.patch('sanctions.apps.api_client.sdn_client.SDNClient.search')
+    def test_sdn_check_accepts_request_without_city(
+        self,
+        mock_search,
+        mock_fallback,
+    ):
+        """Verify that omitting the optional city field still succeeds."""
+        mock_search.return_value = {'total': 0}
+        post_data = {
```
Comment (christopappas) [inline sanctions/apps/api/v1/tests/test_views.py]: you should modify `self.post_data` for this test instead of making a new dictionary here.

File: sanctions/apps/api/v1/tests/test_views.py [hunk 2]
```diff
@@ -119,3 +119,32 @@ def test_sdn_check_search_succeeds_despite_DB_issue(
         assert response.json()['sanctions_check_failure_id'] is None
 
         assert SanctionsCheckFailure.objects.count() == 0
+
+    @mock.patch('sanctions.apps.api.v1.views.checkSDNFallback')
+    @mock.patch('sanctions.apps.api_client.sdn_client.SDNClient.search')
+    def test_sdn_check_accepts_request_without_city(
```
Comment (christopappas) [inline sanctions/apps/api/v1/tests/test_views.py:125]: we should also add a test for when `city` is handed to us, but is an empty string

File: sanctions/apps/api_client/sdn_client.py [hunk 3]
```diff
@@ -36,15 +36,24 @@ def search(self, lms_user_id, name, city, country):
...
             # We are using the city as the address parameter value as indicated in the documentation:
             # https://internationaltradeadministration.github.io/developerportal/consolidated-screening-list.html
-            'city': str(city).encode('utf-8'),
-            'countries': country
-        }
+            params_dict['city'] = str(city).encode('utf-8')
+
         params = urlencode(params_dict)
         sdn_check_url = '{api_url}?{params}'.format(
             api_url=self.sdn_api_url,
```
Comment (christopappas) [inline sanctions/apps/api_client/sdn_client.py:59]: just a note that even though we (sanctions) accepts city as optional, we'll still want/need to check that the gov API accepts no city. given that unit tests mock out real call, before this ships and/or before it reaches production, we'll want to verify the call from sanctions->gov is successful

Comment (colinbrash) [reply sanctions/apps/api_client/sdn_client.py:59]: agreed, i'd like someone to test this.  the docs at least do claim it is not required: https://developer.trade.gov/api-details#api=consolidated-screening-list&operation=search

File: sanctions/apps/api/v1/tests/test_views.py (hunk 1 above)
Comment (colinbrash) [reply sanctions/apps/api/v1/tests/test_views.py]: done

File: sanctions/apps/api/v1/tests/test_views.py (hunk 2 above)
Comment (colinbrash) [reply sanctions/apps/api/v1/tests/test_views.py:125]: done

File: sanctions/apps/api_client/sdn_client.py (hunk 3 above)
Comment (shafqatfarhan) [reply sanctions/apps/api_client/sdn_client.py:59]: I can confirm that the GOV API works without the city parameter.

**Review by shafqatfarhan — APPROVED** [review]
